status_map/__init__.py

```diff
@@ -18,7 +19,8 @@ class StatusMap(Mapping):
     def __init__(self, transitions):
         graph = DiGraph()
         graph.add_nodes_from(transitions.keys())
-        for node in graph.nodes:
+        nodes = deepcopy(graph.nodes)
```

Comment:
can you add a unit test to expose the fix?

Reply:
since the error is hard to reproduce it's even harder to create a test to it. the only thing we could test is whether the structures are different but that doesn't say much. i.e. I don't see much advantage on creating tests exactly for this.